src/libraries/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs

```diff
@@ -803,13 +803,10 @@ public HttpListenerContext EndGetContext(IAsyncResult asyncResult)
                 if (authorizationHeader != null && (authenticationScheme & ~AuthenticationSchemes.Anonymous) != AuthenticationSchemes.None)
                 {
                     // Find the end of the scheme name.  Trust that HTTP.SYS parsed out just our header ok.
-                    for (index = 0; index < authorizationHeader.Length; index++)
+                    index = authorizationHeader.AsSpan().IndexOfAny(" \t\r\n");
+                    if (index < 0)
                     {
-                        if (authorizationHeader[index] == ' ' || authorizationHeader[index] == '\t' ||
-                            authorizationHeader[index] == '\r' || authorizationHeader[index] == '\n')
-                        {
-                            break;
-                        }
+                        index = authorizationHeader.Length;
```

Comment:
This will work. But I think this is only used in the condition bellow to guard if any character was found. e.g. you can simplify that without assignment as 

```diff
-  if (index < authorizationHeader.Length)
+  if (index >= 0)
```

Reply:
yeah, I missed that reading it on GH. However, without going to the first block, headerScheme should remain `AuthenticationSchemes.None` and we will error out without reaching the next `index` use. Not sure if HttpListener is worth of any more troubles.